sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceivedMessageTest.java

```diff
@@ -75,24 +96,53 @@ public void toServiceBusMessageTest() {
+
+        originalMessageAnnotations.put(SEQUENCE_NUMBER_ANNOTATION_NAME.getValue(), Long.valueOf(3));
+
+        final Map<String, Object> originalApplicationProperties = originalMessage.getAmqpAnnotatedMessage().getApplicationProperties();
+        originalApplicationProperties.put(DEAD_LETTER_DESCRIPTION_ANNOTATION_NAME.getValue(), "description");
+        originalApplicationProperties.put(DEAD_LETTER_REASON_ANNOTATION_NAME.getValue(), "description");
+
+        originalMessage.getAmqpAnnotatedMessage().getHeader().setDeliveryCount(Long.valueOf(5));
 
         // Act
-        final ServiceBusMessage messageToSend = new ServiceBusMessage(originalMessage);
+        final ServiceBusMessage actual = new ServiceBusMessage(originalMessage);
 
         // Assert
-        Assertions.assertNotNull(messageToSend);
-        Assertions.assertNotNull(messageToSend.getBody());
-        Assertions.assertEquals(PAYLOAD, new String(messageToSend.getBody(), UTF_8));
-        Assertions.assertEquals(originalMessage.getMessageId(), messageToSend.getMessageId());
-        Assertions.assertEquals(originalMessage.getContentType(), messageToSend.getContentType());
-        Assertions.assertEquals(originalMessage.getCorrelationId(), messageToSend.getCorrelationId());
-        Assertions.assertEquals(originalMessage.getReplyTo(), messageToSend.getReplyTo());
-        Assertions.assertEquals(originalMessage.getViaPartitionKey(), messageToSend.getViaPartitionKey());
-        Assertions.assertEquals(originalMessage.getTimeToLive().toMillis(), messageToSend.getTimeToLive().toMillis());
-        Assertions.assertEquals(originalMessage.getLabel(), messageToSend.getLabel());
-        Assertions.assertEquals(originalMessage.getReplyToSessionId(), messageToSend.getReplyToSessionId());
-        Assertions.assertEquals(originalMessage.getTo(), messageToSend.getTo());
+        assertNotNull(actual);
+        assertNotNull(actual.getBody());
+        assertEquals(PAYLOAD, new String(actual.getBody(), UTF_8));
+        assertEquals(originalMessage.getMessageId(), actual.getMessageId());
+        assertEquals(originalMessage.getContentType(), actual.getContentType());
+        assertEquals(originalMessage.getCorrelationId(), actual.getCorrelationId());
+        assertEquals(originalMessage.getReplyTo(), actual.getReplyTo());
+        assertEquals(originalMessage.getViaPartitionKey(), actual.getViaPartitionKey());
+        assertEquals(originalMessage.getTimeToLive().toMillis(), actual.getTimeToLive().toMillis());
+        assertEquals(originalMessage.getLabel(), actual.getSubject());
+        assertEquals(originalMessage.getReplyToSessionId(), actual.getReplyToSessionId());
+        assertEquals(originalMessage.getTo(), actual.getTo());
+
+        // Following values should be cleaned up.
+        assertNullValues(actual.getAmqpAnnotatedMessage().getMessageAnnotations(), DEAD_LETTER_SOURCE_KEY_ANNOTATION_NAME,
```

Comment:
Do we need a method for this? if we do assertNull, we won't understand which key failed, just that one of the keys was null when we did not want it to be. If we can split it out and reason about the lines